vpn.py

```diff
@@ -64,23 +68,76 @@ def scan_vpns(directory, auth_file):
         # to run
         config = centinel.config.Configuration()
         config.parse_config(centinel_config)
+
+        # assuming that each VPN config file has a name like:
+        # [ip-address].ovpn, we can extract IP address from filename
+        # and use it to geolocate and fetch experiments before connecting
+        # to VPN.
+        vpn_address, extension = os.path.splitext(filename)
+        country = None
+        try:
+            geo = centinel.backend.geolocate(config.params,
```

Comment:
On average, a single run on non-US endpoints with minimal baseline experiments takes ~12 hours. It will take much longer when run with actual URL lists (O(1000) URLs per client).
So no, I don't think running this once per hour is a good idea. We're still benchmarking it to figure out how often it should be run.